### PR TITLE
SPT-6030: Looking up an app/record/user/group/etc, by name/id/refID/etc fail if the value being used is empty or None

### PR DESCRIPTION
**Pull request by danguloa**
- For the app GET by id and name call, check if lookup value is empty or None, if it is raises a ValueError and returns a specific response
- For the records GET by id and tracking_id call, check if lookup value is empty or None, if it is raises a ValueError and returns a specific response
- For the users GET by id and display_name call, check if lookup value is empty or None, if it is raises a ValueError and returns a specific response
- For the groups GET by id and name call, check if lookup value is empty or None, if it is raises a ValueError and returns a specific response
- Update and add tests to app, records, users and groups adapters

### CODE REVIEW

**Review by jeanmurillo95 — APPROVED** [review]
Approved

File: swimlane/core/adapters/record.py [hunk 1]
```diff
@@ -31,7 +31,12 @@ def get(self, key, value):
 
         Raises:
             TypeError: No id argument provided
+            ValueError: The lookup value is empty or None
         """
+
+        if not value:
+            raise ValueError('No {} provided'.format(key))
```
Comment (seandgrimes) [inline swimlane/core/adapters/record.py:38]: Same as previous comment

File: swimlane/core/adapters/usergroup.py [hunk 2]
```diff
@@ -60,10 +60,14 @@ def get(self, key, value):
         Raises:
             TypeError: Unexpected or more than one keyword argument provided
             ValueError: No matching group found based on provided inputs
+            ValueError: The lookup value is empty or None
 
         Returns:
             Group: Group instance matching provided inputs
         """
+        if not value:
+            raise ValueError('No {} provided'.format(key))
```
Comment (seandgrimes) [inline swimlane/core/adapters/usergroup.py:69]: Same as previous comment

File: swimlane/core/adapters/usergroup.py [hunk 3]
```diff
@@ -138,7 +142,11 @@ def get(self, arg, value):
         Raises:
             TypeError: Unexpected or more than one keyword argument provided
             ValueError: No matching user found based on provided inputs, or multiple Users with same display name
+            ValueError: The lookup value is empty or None
         """
+        if not value:
+            raise ValueError('No {} provided'.format(arg))
```
Comment (seandgrimes) [inline swimlane/core/adapters/usergroup.py:148]: Same as previous comment

File: swimlane/core/adapters/app.py [hunk 4]
```diff
@@ -24,7 +24,11 @@ def get(self, key, value):
         Raises:
             TypeError: No or multiple keyword arguments provided
             ValueError: No matching app found on server
+            ValueError: The lookup value is empty or None
         """
+        if not value:
+            raise ValueError('No {} provided'.format(key))
```
Comment (seandgrimes) [inline swimlane/core/adapters/app.py:30]: Would it be any clearer to use something like `'The value provided for the key "{0}" cannot be empty or None'` here?

**Review by seandgrimes — CHANGES_REQUESTED** [review]

**Review by seandgrimes — APPROVED** [review]

**Review by seandgrimes — APPROVED** [review]

**Pull request merged**